packages/renderer/src/lib/compose/ComposeActions.spec.ts

```diff
@@ -103,13 +105,18 @@ test('Expect no error and status restarting compose', async () => {
 });
 
 test('Expect no error and status deleting compose', async () => {
+  // Mock the showMessageBox to return 0 (yes)
+  showMessageBoxMock.mockResolvedValue({ response: 0 });
```

Comment:
I really like what handleConfirmationDialog() does for the e2e tests, and that's kind of what I was hoping for in the regular tests, i.e. is there any way to change the pattern of:
   showMessageBoxMock/`await fireEvent.click(deleteButton)`
to something like::
   `await utility.clickButtonAndConfirm(deleteButton)`
that would do the click and respond yes (or mock/disable the confirmation dialog completely)? (and no, I don't know of/think we have a common place for things like this yet)

Reply:
Due to https://vitest.dev/guide/mocking.html#mocking-pitfalls

The 'screen' / rendering portion stays within the test() function, so we're unable to move this to a separate file for mocking / using screen to do clicking.

I've tried multiple ways to implement this into a separate function, but the screen does not "transfer" over, even when passing in `screen: any` to the function.

```ts
export async function clickAndConfirm(clickAction: () => void) {
  const showMessageBoxMock = vi.fn();
  (window as any).showMessageBox = showMessageBoxMock;
  showMessageBoxMock.mockResolvedValue({ result: 1 });
  clickAction();
  await waitFor(() => expect(screen.queryByRole('dialog')).not.toBeInTheDocument());
}
```

Reply:
ok, thanks for investigating.